fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsBottomSheet.kt

```diff
@@ -0,0 +1,147 @@
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer2,
+                shape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp),
+            )
+            .nestedScroll(rememberNestedScrollInteropConnection()),
+    ) {
+        androidx.compose.material.Divider(
+            Modifier
+                .padding(top = 16.dp)
+                .fillMaxWidth(BOTTOM_SHEET_HANDLE_WIDTH_PERCENT)
+                .align(alignment = Alignment.CenterHorizontally),
+            color = FirefoxTheme.colors.borderInverted,
+            thickness = 3.dp,
+        )
+
+        content()
+    }
+}
+
+@Composable
+internal fun TranslationsAnimation(
+    translationsVisibility: Boolean,
+    density: Density,
+    translationsOptionsHeightDp: Dp,
+    content: @Composable AnimatedVisibilityScope.() -> Unit,
+) {
+    AnimatedVisibility(
+        visible = translationsVisibility,
+        enter = expandIn(
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+            initialSize = {
+                with(density) {
+                    IntSize(
+                        0,
+                        translationsOptionsHeightDp.roundToPx(),
+                    )
+                }
+            },
+        ) + fadeIn(
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+        ) + slideInHorizontally(
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+        ),
+    ) {
+        content()
+    }
+}
+
+@Composable
+internal fun TranslationsOptionsAnimation(
+    translationsVisibility: Boolean,
+    density: Density,
+    translationsHeightDp: Dp,
+    translationsWidthDp: Dp,
+    content: @Composable AnimatedVisibilityScope.() -> Unit,
+) {
+    AnimatedVisibility(
+        visible = translationsVisibility,
+        enter = expandIn(
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+            initialSize = {
+                with(density) {
+                    IntSize(
+                        0,
+                        translationsHeightDp.roundToPx(),
+                    )
+                }
+            },
+        ) + fadeIn(
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+        ) + slideInHorizontally(
+            initialOffsetX = { with(density) { translationsWidthDp.roundToPx() } },
+            animationSpec = tween(
+                easing = FastOutSlowInEasing,
+            ),
+        ),
+    ) {
+        content()
+    }
+}
+
+@Composable
+internal fun TranslationsDialog(onSettingClicked: () -> Unit) {
+    TranslationsDialogBottomSheet(
+        onSettingClicked = onSettingClicked,
+    )
+}
+
+@Composable
+internal fun TranslationsOptionsDialog(onBackClicked: () -> Unit, navController: NavController) {
```

Comment:
We shouldn't pass `NavController` here. We should just do a callback.

Reply:
Done